src/file/info.cr

```diff
@@ -0,0 +1,107 @@
+require "c/sys/stat"
+
+class File
+  # Represents the various behaviour-altering flags which can be set on files.
+  # Not all flags will be supported on all platforms.
+  @[Flags]
+  enum Flags : UInt8
+    SetUser
+    SetGroup
+    Sticky
+  end
+
+  # Represents the type of a file. Not all types will be supported on all
+  # platforms.
+  enum Type : UInt8
+    File
+    Directory
+    SymbolicLink
+    Socket
+    Pipe
+    CharacterDevice
+    BlockDevice
+
+    # Returns true if `self` is a `CharacterDevice` or a `BlockDevice`.
+    def device?
+      character_device? || block_device?
+    end
+  end
+
+  # Represents a set of access permissions for a file. Not all permission sets
+  # will be supported on all platforms.
+  #
+  # The binary representation of this enum is defined to be same representation
+  # as the permission bits of a unix `st_mode` field. `File::Permissions`
+  # can also be compared to it's underlying bitset, for example
+  # `File::Permissions::All == 0o777` will always be `true`.
```

Comment:
What about the opposite? Does `0o777 == File::Permissions::All` work?

Reply:
Also, maybe `always` isn't neccessary here.

Reply:
Ah, yeah. Perhaps this equality was a bad idea. Or should I just reopen `struct Int`? It's a bit ugly how these equalities are defined in two places.